File: Valid Compressed String.cpp
class Solution {
  public:
    int checkCompressed(string s, string t) {
        int n = 0;
        int flag = 1;
        int j = 0;
        for(int i=0; i<t.length(); i++)
        {
            if (t[i]>='0' && t[i]<='9')
            {
                n *= 10;
                if(n>100000)
                    return 0;
                n += t[i] - '0';
                j--;
            }
            else
            {
                j += n;
                if(t[i]!=s[j])
                {
                    flag = 0;
                    break;
                }
                n = 0;
            }
            j++;
        }
        j += n;
        if(j != s.length())
            flag = 0;

        if(flag)
            return 1;
        else
            return 0;
    }
};

File: Valid Compressed String.py
class Solution:
    def checkCompressed(self, s, t):
        n = 0
        flag = 1
        j = 0
        i = 0
        while i<len(t):
            if t[i].isnumeric():
                n *= 10
                if n>100000:
                    return 0
                n += int(t[i])
                j -= 1
            else:
                j += n
                if (t[i]!=s[j]):
                    flag = 0
                    break
                n = 0
            j += 1
            i += 1
        j += n
        if(j != len(s)):
            flag = 0
        if(flag):
            return 1
        else:
            return 0
